### PR TITLE
sys_usbd: Guitar Hero Live controller emulation

### PR DESCRIPTION
**Pull request by InvoxiPlayGames**
This adds emulation of the Guitar Hero Live guitar controller using inputs from controller 7, using default bindings for the Xbox 360 Guitar Hero Live guitar.

Not 100% accurate as of yet, mainly (unused in-game) d-pad and control packets.

### CODE REVIEW

**Comment by Megamouse** [comment]
Why GHLtar. I wouldn't know what that is supposed to be if I hadn't read it here

File: rpcs3/Emu/Io/GHLtar.cpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	{
+	// HID Host 2 Device
+	case 0x21:
+		switch (bRequest)
+		{
+		case 0x09: break;
+		default: ghltar_log.error("Unhandled Query Type: 0x%02X", buf[0]); break;
+		}
+		break;
+		break;
```
Comment (Megamouse) [inline rpcs3/Emu/Io/GHLtar.cpp]: Why the double break

File: rpcs3/Emu/Io/GHLtar.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+usb_device_ghltar::~usb_device_ghltar()
+{
+}
+
+void usb_device_ghltar::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue, u16 wIndex, u16 wLength, u32 buf_size, u8* buf, UsbTransfer* transfer)
+{
+	transfer->fake = true;
+
+	// Control transfers are nearly instant
+	switch (bmRequestType)
```
Comment (Megamouse) [inline rpcs3/Emu/Io/GHLtar.cpp]: I don't like how all of the code inside of this is basically pressed into one liners

File: rpcs3/Emu/Io/GHLtar.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	buf[22] = 0x01; // Unknown, always 0x01
+	buf[24] = 0x02; // Unknown, always 0x02
+	buf[26] = 0x02; // Unknown, always 0x02
+	// buf[7] through buf[18] are always 0x00
+	// buf[21]/[23]/[25] are also always 0x00
+
+	const auto handler = pad::get_current_handler();
+	const auto& pads   = handler->GetPads();
+	const auto pad     = pads[6];
+	if (!(pad->m_port_status & CELL_PAD_STATUS_CONNECTED))
```
Comment (Megamouse) [inline rpcs3/Emu/Io/GHLtar.cpp]: This whole section doesn't have any vertical spacing

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 4]
```diff
@@ -168,7 +170,12 @@ usb_handler_thread::usb_handler_thread()
 		check_device(0x1415, 0x0000, 0x0000, "Singstar Microphone");
 		check_device(0x12BA, 0x0100, 0x0100, "Guitar Hero Guitar");
 		check_device(0x12BA, 0x0120, 0x0120, "Guitar Hero Drums");
-		check_device(0x12BA, 0x074B, 0x074B, "Guitar Hero Live Guitar");
+
+		if (check_device(0x12BA, 0x074B, 0x074B, "Guitar Hero Live Guitar"))
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp]: Might just assign this instead of using an if

File: rpcs3/Emu/Io/GHLtar.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
+﻿#include "stdafx.h"
+#include "GHLtar.h"
+#include "Emu/Cell/lv2/sys_usbd.h"
+#include "Input/pad_thread.h"
+#include <Emu\Cell\Modules\cellPad.cpp>
+#include <Emu\RSX\GL\GLHelpers.h>
```
Comment (JohnHolmesII) [inline rpcs3/Emu/Io/GHLtar.cpp]: Don't mix the quotes and the chevrons. Don't mix slashes either. This is why it won't pass CI I'm p sure

Comment (InvoxiPlayGames) [reply rpcs3/Emu/Io/GHLtar.cpp]: ah that makes sense - I just chose the defaults that Visual Studio offered me so I'll go and fix that right now

File: rpcs3/Emu/Io/GHLtar.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿#pragma once
+
+#include "Emu/Io/usb_device.h"
+#include <queue>
+#include <mutex>
```
Comment (Megamouse) [inline rpcs3/Emu/Io/GHLtar.h]: are queue and mutex really needed?

File: rpcs3/Emu/Io/GHLtar.cpp [hunk 7]
```diff
@@ -0,0 +1,215 @@
+﻿// Guitar Hero Live controller emulator
+
+#include "stdafx.h"
+#include "GHLtar.h"
+#include "Emu/Cell/lv2/sys_usbd.h"
+#include "Input/pad_thread.h"
+#include "Emu/Cell/Modules/cellPad.cpp"
+#include "Emu/RSX/GL/GLHelpers.h"
```
Comment (Megamouse) [inline rpcs3/Emu/Io/GHLtar.cpp]: what are these even for?
```
#include "Emu/Cell/Modules/cellPad.cpp"
#include "Emu/RSX/GL/GLHelpers.h"
```

Comment (InvoxiPlayGames) [reply rpcs3/Emu/Io/GHLtar.cpp]: Accidentally left those in when experimenting how to get pad inputs

File: rpcs3/Emu/Io/GHLtar.h (hunk 6 above)
Comment (InvoxiPlayGames) [reply rpcs3/Emu/Io/GHLtar.h]: Just kinda copied from Skylander.h and forgot

**Comment by Megamouse** [comment]
Needs a rebase

**Pull request merged**